### PR TITLE
SERVICES-966 rename elrond-data to elrond-tools; compute historical U…

### PR DESCRIPTION
**Pull request by johnykes**
- elrond-data.service.ts -> elrond-tools.service.ts (also remove unused method)
- get token price from our DEX history via tools.elrond.com
- compute any token price by amount and cached token price

### CODE REVIEW

File: src/common/services/elrond-communication/elrond-tools.service.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+import { elrondConfig } from 'src/config';
+import axios, { AxiosRequestConfig } from 'axios';
+import { ApiConfigService } from 'src/utils/api.config.service';
+import * as Agent from 'agentkeepalive';
+import { PerformanceProfiler } from '@elrondnetwork/erdnest';
+import { MetricsCollector } from 'src/modules/metrics/metrics.collector';
+import { NativeAuthSigner } from '@elrondnetwork/erdnest/lib/src/utils/native.auth.signer';
+import BigNumber from 'bignumber.js';
+
+@Injectable()
```
Comment (danielailie) [inline src/common/services/elrond-communication/elrond-tools.service.ts:11]: maybe use the api.service it already has all axios configs

File: src/utils/date-utils.ts [hunk 2]
```diff
@@ -34,4 +34,18 @@ export class DateUtils {
   static getDateFromTimestampWithoutTime(timestamp: number) {
     return new Date(timestamp * 1000).toJSON().slice(0, 10);
   }
+
+  static timestampToIsoStringWithoutTime(timestamp: number): string {
```
Comment (danielailie) [inline src/utils/date-utils.ts:38]: please rename d to date

**Review by danielailie — CHANGES_REQUESTED** [review]

File: src/utils/date-utils.ts (hunk 2 above)
Comment (johnykes) [reply src/utils/date-utils.ts:38]: Done

File: src/common/services/elrond-communication/elrond-tools.service.ts (hunk 1 above)
Comment (johnykes) [reply src/common/services/elrond-communication/elrond-tools.service.ts:11]: Done

File: src/common/services/elrond-communication/elrond-tools.service.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+      const response = await this.apiService.post(this.url, { query }, config);
+      return response.data;
+    } catch (error) {
+      this.logger.error(`Error when trying to get run ${name}`, {
+        error: error.message,
+        path: `${ElrondToolsService.name}.${this.doPost.name}`,
+      });
+    } finally {
+      profiler.stop();
```
Comment (danielailie) [inline src/common/services/elrond-communication/elrond-tools.service.ts:103]: The profiling is already done in api service no need to do it also here

Comment (johnykes) [reply src/common/services/elrond-communication/elrond-tools.service.ts]: Done

**Review by danielailie — APPROVED** [review]

**Review by danielailie — APPROVED** [review]

**Review by gabrielmatei — APPROVED** [review]

**Pull request merged**